src/foundry/foundry.js/applications/hotbar.d.ts

```diff
@@ -1,152 +1,163 @@
-   */
-  protected _onHoverMacro(event: JQuery.MouseEnterEvent | JQuery.MouseLeaveEvent): void;
-
-  /**
-   * Handle pagination controls
-   * @param event - The originating click event
-   */
-  protected _onClickPageControl(event: JQuery.ClickEvent): void;
-
-  /** @override */
-  protected _canDragStart(selector: string | null): boolean;
-
-  /** @override */
-  protected _canDragDrop(selector: string | null): boolean;
-
-  /** @override */
-  protected _onDrop(event: DragEvent): void;
-
-  /**
-   * Get the Macro entity being dropped in the Hotbar. If the data comes from a non-World source, create the Macro
-   * @param data - The data transfer attached to the DragEvent
-   * @returns A Promise which returns the dropped Macro, or null
-   */
-  protected _getDropMacro(data: Hotbar.DropData): Promise<Macro | null>;
-
-  /**
-   * Handle click events to toggle display of the macro bar
-   * @param event -
-   */
-  protected _onToggleBar(event: JQuery.ClickEvent): void;
-}
-
-declare namespace Hotbar {
-  interface Data {
+import { ConfiguredDocumentClass } from '../../../types/helperTypes';
+
+declare global {
+  /**
+   * The global action bar displayed at the bottom of the game view.
+   * The Hotbar is a UI element at the bottom of the screen which contains Macros as interactive buttons.
+   * The Hotbar supports 5 pages of global macros which can be dragged and dropped to organize as you wish.
+   *
+   * Left clicking a Macro button triggers its effect.
+   * Right clicking the button displays a context menu of Macro options.
+   * The number keys 1 through 0 activate numbered hotbar slots.
+   * Pressing the delete key while hovering over a Macro will remove it from the bar.
+   *
+   * @see {@link Macros}
+   * @see {@link Macro}
+   *
+   * @typeParam Options - the type of the options object
+   */
+  class Hotbar<Options extends Application.Options = Application.Options> extends Application<Options> {
+    constructor(options?: Partial<Options>);
+
+    /**
+     * The currently viewed macro page
+     * @defaultValue `1`
+     */
     page: number;
-    macros: Macro[];
-    barClass: 'collapsed' | '';
+
+    /**
+     * The currently displayed set of macros
+     * @defaultValue `[]`
+     */
+    macros: InstanceType<ConfiguredDocumentClass<typeof Macro>>[];
+
+    /**
+     * Track collapsed state
+     * @defaultValue `false`
+     */
+    protected _collapsed: boolean;
+
+    /**
+     * Track which hotbar slot is the current hover target, if any
+     * @defaultValue `null`
+     */
+    protected _hover: number | null;
+
+    /**
+     * @override
+     * @defaultValue
+     * ```typescript
+     * mergeObject(super.defaultOptions, {
+     *   id: "hotbar",
+     *   template: "templates/hud/hotbar.html",
+     *   popOut: false,
+     *   dragDrop: [{ dragSelector: ".macro-icon", dropSelector: "#macro-list" }]
+     * })
+     * ```
+     */
+    static get defaultOptions(): typeof Application['defaultOptions'];
```

Comment:
```suggestion
    static get defaultOptions(): Application.Options;
```
That's what we typically use. It's the same type and doesn't matter much, but this is a bit more consistent